### PR TITLE
Add coverage tests for CSV format, provider artifact sharing, and VAPID store

### PR DESCRIPTION
**Pull request by curdriceaurora**
## Summary
- add focused tests for CSV formatting behavior
- add provider artifact sharing test coverage
- add VAPID store test coverage, including persistence and edge cases
- move advanced settings doc under `docs/` and align related script/package references

## Notes
- this PR includes the existing branch commits

### CODE REVIEW

File: tests/vapid-store.test.js [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    expect(result.subject).toBe("https://example.com");
+  });
+});
+
+describe("VapidStore constructor", () => {
+  test("requires filePath option", () => {
+    expect(() => new VapidStore()).toThrow(VapidStoreError);
+    expect(() => new VapidStore({})).toThrow(VapidStoreError);
+    expect(() => new VapidStore({ filePath: "/tmp/keys.json" })).not.toThrow();
+  });
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply tests/vapid-store.test.js:104]: Updated this test to build the file path with `path.join(os.tmpdir(), "keys.json")` instead of a hard-coded `/tmp` path, so it stays portable across Windows and Unix environments.


_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._
